src/components/checkbox/checkbox.scss

```diff
@@ -0,0 +1,401 @@
+@import "default-theme";
+@import "theme-functions";
+@import "variables";
+
+/** The width/height of the checkbox element. */
+$md-checkbox-size: 18px !default;
+/** The width of the line used to draw the checkmark / mixedmark. */
+$md-checkbox-mark-stroke-size: 2/15 * $md-checkbox-size !default;
+/** The width of the checkbox border shown when the checkbox is unchecked. */
+$md-checkbox-border-width: 2px;
+/** The color of the checkbox border. */
+$md-checkbox-border-color: if($md-is-dark-theme, rgba(white, 0.7), rgba(black, 0.54)) !default;
+/** The color of the checkbox's checkmark / mixedmark. */
+$md-checkbox-mark-color: map-get($md-background, background);
+/** The color that is used as the checkbox background when it is checked. */
+$md-checkbox-background-color: md-color($md-primary, 500);
+/** The base duration used for the majority of transitions for the checkbox. */
+$md-checkbox-transition-duration: 140ms;
+/** The amount of spacing between the checkbox and its label. */
+$md-checkbox-item-spacing: 4px;
+
+// Manual calculation done on SVG
+$_md-checkbox-mark-path-length: 22.910259;
+$_md-checkbox-indeterminate-checked-easing-function: cubic-bezier(0.14, 0, 0, 1);
+
+@keyframes fade-in-background {
```

Comment:
Should the keyframe animation names be prefixed with `md-` in order to avoid collisions?